src/components/textAreas/textAreaRichText/textAreaRichText.tsx

```diff
@@ -0,0 +1,115 @@
+import { Link } from '@tiptap/extension-link';
+import { Placeholder } from '@tiptap/extension-placeholder';
+import { EditorContent, useEditor } from '@tiptap/react';
+import { StarterKit } from '@tiptap/starter-kit';
+import classNames from 'classnames';
+import { useEffect, useId, useState } from 'react';
+import { createPortal } from 'react-dom';
+import { InputContainer, type IInputContainerProps } from '../../input';
+import { TextAreaRichTextActions } from './textAreaRichTextActions';
+
+export interface ITextAreaRichTextProps
+    extends Omit<IInputContainerProps, 'maxLength' | 'inputLength' | 'value' | 'onChange' | 'id'> {
+    /**
+     * Current value of the input.
+     */
+    value?: string;
+    /**
+     * Id of the input.
+     */
+    id?: string;
+    /**
+     * Callback called on value change.
+     */
+    onChange?: (value: string) => void;
+    /**
+     * Placeholder of the input.
+     */
+    placeholder?: string;
+}
+
+// Classes to properly style the TipTap placeholder
+// (see https://tiptap.dev/docs/editor/api/extensions/placeholder#additional-setup)
+const placeholderClasses = classNames(
+    'first:before:pointer-events-none first:before:absolute first:before:top-5', // General
+    'first:before:text-base/tight first:before:font-normal first:before:text-neutral-300', // Typography
+    'first:before:content-[attr(data-placeholder)]', // Content
+);
+
+export const TextAreaRichText: React.FC<ITextAreaRichTextProps> = (props) => {
+    const { value, onChange, placeholder, isDisabled, className, id, ...containerProps } = props;
+
+    const [isExpanded, setIsExpanded] = useState(false);
+
+    // Use random-id when id property is not specified for the input.
+    const randomId = useId();
+    const processedId = id ?? randomId;
+
+    const extensions = [
+        StarterKit,
+        Placeholder.configure({ placeholder, emptyNodeClass: placeholderClasses, showOnlyWhenEditable: false }),
+        Link,
+    ];
+
+    const editor = useEditor({
+        extensions,
+        content: value,
+        editable: !isDisabled,
+        editorProps: {
+            attributes: {
```

Comment:
Can we get the text to wrap horizontally? Right now because it doesn't have a width, we can write forever horizontally. 
In the same vein, do you know in terms of design if we want this to have a fixed height/number of lines like a native textarea or should it grow vertically with the content?

Reply:
Fixed first point, will ask Selim for second point for both this and the text-area component 👌🏻